src/Tes/Repository/BlockBlobDatabase.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Identity;
+using Azure.Storage.Blobs;
+using Polly;
+
+namespace Tes.Repository
+{
+    public class BlockBlobDatabase<T> where T : class
+    {
+        private const int maxConcurrentItemDownloads = 64;
+        private const string activeStatePrefix = "a/";
+        private const string inactiveStatePrefix = "z/";
+
+        private readonly BlobServiceClient blobServiceClient;
+        private readonly BlobContainerClient container;
+
+        public string StorageAccountName { get; set; }
+        public string ContainerName { get; set; } 
+
+        public BlockBlobDatabase(string storageAccountName, string containerName, string containerSasToken = null)
+        {
+            StorageAccountName = storageAccountName;
+            ContainerName = containerName;
+
+            if (!string.IsNullOrWhiteSpace(containerSasToken))
+            {
+                blobServiceClient = new BlobServiceClient(new Uri($"https://{StorageAccountName}.blob.core.windows.net?{containerSasToken.TrimStart('?')}"));
+            }
+            else
+            {
+                // Use managed identity. Token lifetime and refreshing is handled automatically.
+                blobServiceClient = new BlobServiceClient(new Uri($"https://{StorageAccountName}.blob.core.windows.net"), new DefaultAzureCredential());
+            }
+
+            container = blobServiceClient.GetBlobContainerClient(ContainerName);
+            container.CreateIfNotExistsAsync().Wait();
+        }
+
+        public async Task CreateOrUpdateItemAsync(string id, T item, bool isActive)
+        {
+            var json = System.Text.Json.JsonSerializer.Serialize(item);
+
+            if (!isActive)
+            {
+                // Delete active if it exists
+                var blobClient1 = container.GetBlobClient(GetActiveBlobNameById(id));
+                var task1 = blobClient1.DeleteIfExistsAsync();
+
+                // Update/create active
+                var blobClient2 = container.GetBlobClient(GetInactiveBlobNameById(id));
+                var task2 = blobClient2.UploadAsync(BinaryData.FromString(json), overwrite: true);
+
+                // Retry to reduce likelihood of one blob succeeding and the other failing
+                await Policy
```

Comment:
Hmm... I think this retry policy could lead to data loss. Two requests roughly at the same time hit this method, the first fails and the second succeeds (we want the second one to win), the first request will go in retry loop that could overwrite the latest data. 

As an alternative you can consider an optimistic concurrency approach for the update, where you check if an item has changed since you last read it - storage supports this via headers, and etags.  And for the create scenario, you turn overwrite off,  to avoid any race condition.